libs/orders/src/lib/components/order-list-manager/order-list-manager.tsx

```diff
@@ -96,27 +80,84 @@ export const OrderListManager = ({
     [create]
   );
 
+  const [hasFilteredRows, setHasFilteredRowsCount] = useState<
+    boolean | undefined
+  >(undefined);
+  const { onFilterChanged, ...props } = gridProps || {};
+  const onRowDataUpdated = useCallback(
+    ({ api }: { api: AgGridReact['api'] }) => {
+      let hasFilteredRows = false;
+      try {
+        api.forEachNodeAfterFilter(() => {
+          throw new Error();
+        });
+      } catch (e) {
+        hasFilteredRows = true;
+      }
+      setHasFilteredRowsCount(hasFilteredRows);
+    },
+    []
+  );
+
   if (error) {
     return <Splash>{t(`Something went wrong: ${error.message}`)}</Splash>;
   }
 
   return (
     <>
       <div className="h-full relative">
-        <OrderListTable
-          rowData={data}
-          ref={gridRef}
-          filter={filter}
-          onCancel={cancel}
-          onEdit={setEditOrder}
-          onView={setViewOrder}
-          onMarketClick={onMarketClick}
-          onOrderTypeClick={onOrderTypeClick}
-          isReadOnly={isReadOnly}
-          overlayNoRowsTemplate={noRowsMessage || t('No orders')}
-          {...gridProps}
-          onFilterChanged={onFilterChanged}
-        />
+        <div className="flex flex-col h-full">
+          <OrderListTable
+            rowData={data}
+            ref={gridRef}
+            filter={filter}
+            onCancel={cancel}
+            onEdit={setEditOrder}
+            onView={setViewOrder}
+            onMarketClick={onMarketClick}
+            onOrderTypeClick={onOrderTypeClick}
+            onFilterChanged={(event) => {
+              onRowDataUpdated(event);
+              if (onFilterChanged) {
+                onFilterChanged(event);
+              }
+            }}
+            onRowDataUpdated={onRowDataUpdated}
+            isReadOnly={isReadOnly}
+            overlayNoRowsTemplate={noRowsMessage || t('No orders')}
+            {...props}
+          />
+          <div className="flex justify-between border-t border-default p-1 items-center">
+            <div className="text-xs">
+              {t(
+                'Depending on data node retention you may not be able see the "full" history'
+              )}
+            </div>
+            {data ? (
+              <div className="flex text-xs items-center">
```

Comment:
Probably worth adding a datagrid or ui-toolkit component for the pagination component

Reply:
I don't think it's necessary. This is just two div elements with classes. Most probably we will be changing it's behaviour depend on place where it's used